CMakeLists.txt

```diff
@@ -157,173 +164,159 @@ set(soci_extra_includes
+        beast/insight
+        beast/net
+        beast/utility
+        app
+        basics
+        crypto
+        json
+        ledger
+        legacy
+        net
+        overlay
+        peerfinder
+        protocol
+        rpc
+        server
+        shamap
+        test)
     file(GLOB_RECURSE cursrcs src/ripple/${curdir}/*.cpp)
-    list(APPEND rippled_src "${cursrcs}")
+    list(APPEND rippled_src_nonunity "${cursrcs}")
     list(APPEND non_unity_srcs "${cursrcs}")
-  endforeach()
+endforeach()
 
-  file(GLOB_RECURSE nodestore_srcs src/ripple/nodestore/*.cpp
+file(GLOB_RECURSE nodestore_srcs src/ripple/nodestore/*.cpp
     src/test/nodestore/*.cpp)
 
-  add_with_props(rippled_src "${nodestore_srcs}"
+add_with_props(rippled_src_nonunity "${nodestore_srcs}"
     -I"${CMAKE_SOURCE_DIR}/"src/rocksdb2/include
     -I"${CMAKE_SOURCE_DIR}/"src/snappy/snappy
     -I"${CMAKE_SOURCE_DIR}/"src/snappy/config
     ${rocks_db_system_header})
-  
-  list(APPEND non_unity_srcs "${nodestore_srcs}")
-  
 
-  file(GLOB_RECURSE test_srcs src/test/*.cpp)
-  add_with_props("${test_srcs}"
+list(APPEND non_unity_srcs "${nodestore_srcs}")
+
+file(GLOB_RECURSE test_srcs src/test/*.cpp)
+add_with_props(rippled_src_nonunity "${test_srcs}"
     -I"${CMAKE_SOURCE_DIR}/"src/rocksdb2/include
     -I"${CMAKE_SOURCE_DIR}/"src/snappy/snappy
     -I"${CMAKE_SOURCE_DIR}/"src/snappy/config
     ${rocks_db_system_header})
-  
-  list(APPEND non_unity_srcs "${test_srcs}")
-  list(APPEND rippled_src "${test_srcs}")
-
-  file(GLOB_RECURSE rippled_headers src/*.h src/*.hpp)
-  foreach(curdir
-      beast/asio
-      beast/core
-      beast/crypto
-      beast/cxx17
-      beast/hash
-      proto
-      resource
-      validators
-      websocket)
-    file(GLOB_RECURSE cursrcs src/ripple/${curdir}/*.cpp)
-    list(APPEND rippled_headers "${cursrcs}")
-  endforeach()
-  list(APPEND rippled_src "${rippled_headers}")
 
-  # Properties
-  set_property(
-    SOURCE ${non_unity_srcs}
-    APPEND
-    PROPERTY HEADER_FILE_ONLY
-    ${unity})
-  set_property(
-    SOURCE ${rippled_headers}
-    APPEND
-    PROPERTY HEADER_FILE_ONLY
-    true)
-    # Doesn't work
-    # $<OR:$<CONFIG:Debug>,$<CONFIG:Release>>)
+list(APPEND non_unity_srcs "${test_srcs}")
+
+if(WIN32 OR is_xcode)
+    # Rippled headers. Only needed for IDEs.
+    file(GLOB_RECURSE rippled_headers src/*.h src/*.hpp)
+    list(APPEND rippled_headers Builds/CMake/CMakeFuncs.cmake)
+    foreach(curdir
+            beast/asio
+            beast/core
+            beast/crypto
+            beast/cxx17
+            beast/hash
+            proto
+            resource
+            validators
+            websocket)
+        file(GLOB_RECURSE cursrcs src/ripple/${curdir}/*.cpp)
+        list(APPEND rippled_headers "${cursrcs}")
+    endforeach()
+    list(APPEND rippled_src_nonunity "${rippled_headers}")
+
+    set_property(
+        SOURCE ${rippled_headers}
+        APPEND
+        PROPERTY HEADER_FILE_ONLY
+        true)
```

Comment:
Doesn't this unconditionally set non-unity source files as header files, even for non-unity builds? I can't (easily) test with an IDE, but that does the right thing in visual studio when the current configuration is a classic build?